### DISCUSSION TITLE
file_id for disk - what is it.

### DISCUSSION THREAD

**Discussion by jeffellin** [bpg/terraform-provider-proxmox#1364] [Q&A] [answered]
The examples in the docs that create a cloud-init vm show downloading an image and then referencing its id in the file_id section.  What is the file_id if I want to use an image I already downloaded?  Is it the path to the file or something else?

The docs just say that the proxmox_virtual_environment_download_file returns an  id (String) The unique identifier of this resource.

**Comment by bpg-dev** [comment, marked as answer]
Hi @jeffellin,

Here is an example of `file_id`: `local:iso/Memtest86-4.3.7.iso`.
Basically it is `<datastore>:<content_type>/<filename>`.

Here is another example from the docs: https://registry.terraform.io/providers/bpg/proxmox/latest/docs/guides/cloud-image#create-a-vm-from-an-exiting-image-on-proxmox

**Comment by bpg-dev** [comment]
Documented the format in https://github.com/bpg/terraform-provider-proxmox/pull/1374
